### PR TITLE
Remove webpack/sjs bundler from jsdocs

### PR DESCRIPTION
**Pull request by armanbilge**
Might need some tweaking but at a glance seems to work.

### CODE REVIEW

File: core/js/src/main/scala/viz/vega/facades/VegaEmbed.scala [hunk 1]
```diff
@@ -83,5 +83,5 @@ trait EmbedOptions:
 object VegaEmbed:
 
   @js.native
-  @JSImport("vega-embed", JSImport.Default)
+  @JSImport("vega-embed", JSImport.Default, "vegaEmbed")
```
Comment (armanbilge) [inline core/js/src/main/scala/viz/vega/facades/VegaEmbed.scala:86]: What this does is import `vega-embed` if modules are enabled. If modules are not enabled, it instead looks for a `vegaEmbed` global. This is compatible with their official docs:

https://vega.github.io/vega/usage/#using-the-vega-embed-module

File: core/js/src/main/scala/viz/vega/facades/VegaView.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ import scala.scalajs.js.annotation.JSImport
 import scala.scalajs.js.JSON
 
 @js.native
-@JSImport("vega-view", JSImport.Namespace)
+@JSImport("vega-view", JSImport.Namespace, "vega.View")
```
Comment (armanbilge) [inline core/js/src/main/scala/viz/vega/facades/VegaView.scala:24]: I'm not sure if `"vega.View"` is right. Are you using this anywhere in your documentation, so we can check that it worked?

File: raw_docs/default.template.html [hunk 3]
```diff
@@ -22,6 +22,9 @@
     @:linkJS { paths = ${helium.site.includeJS} }
     @:heliumInitVersions
     @:heliumInitPreview(container)
+    <script src="https://cdn.jsdelivr.net/npm/vega@5"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-lite@5"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6"></script>
```
Comment (armanbilge) [inline raw_docs/default.template.html:27]: Copied from here: https://vega.github.io/vega/usage/#using-the-vega-embed-module

File: core/js/src/main/scala/viz/vega/facades/VegaView.scala (hunk 2 above)
Comment (armanbilge) [reply core/js/src/main/scala/viz/vega/facades/VegaView.scala:24]: I did a small experiment and I am more confident this is correct now :)

File: core/js/src/main/scala/viz/vega/facades/VegaEmbed.scala (hunk 1 above)
Comment (Quafadas) [reply core/js/src/main/scala/viz/vega/facades/VegaEmbed.scala:86]: That sounds like very attractive behaviour!

**Comment by Quafadas** [comment]
I'll try and have an another look at it again today - I had a quick go and for me I got the same problems ... I'll be looking as soon as I ge the chance :-).

**Comment by Quafadas** [comment]
So, I still get the same error, when going to the docs project, running tlSitePreview, and navigating to the gettingStarted.js page. 

My branch is "changeBundler2". 

I did manage to get rid of the first error, by changing the generated output as follows. In gettingStarted.md.js, from 

`import * as $i_vega$002dembed from "vega-embed";` 
to
`import * as $i_vega$002dembed from "https://cdn.jsdelivr.net/npm/vega-embed@6.21.3/+esm";`

and then in the generated gettingStart.md

from 
`<script type="module" src="gettingStarted.md.js" defer></script>`
to
`<script type="module" src="gettingStarted.md.js" defer></script>`

That appears to correctly reference Vega, however, I don't think mdoc is then wired up to take advantage of it correctly. Leaving me with this problem;
`ReferenceError: Can't find variable: mdoc_js_run0`

**Comment by Quafadas** [comment]
Ridiculously... 

`@JSImport("https://cdn.jsdelivr.net/npm/vega-embed@6.21.3/+esm", JSImport.Default, "vegaEmbed")` actually does appear to solve the first of the above.

**Comment by armanbilge** [comment]
I'm a bit confused what you are doing :) does this branch, as-is, not work for you?

Regarding rewriting your `@JSImport`s: I don't think that makes sense to hard-code a CDN in a published library. I actually just opened a feature request in Scala.js asking for a feature so that we can rewrite these imports at linking time.

- https://github.com/scala-js/scala-js/issues/4837

**Comment by Quafadas** [comment]
I'm currently having a bit of a wtf moment, because your branch works. What I cannot understand, is that I swear I made these changes, and mine don't work. I get different 

Which super duper sucks. I suspect, that this line in build.sbt causes a mess...

`
ThisBuild /  scalaJSLinkerConfig ~= (
   _.withModuleKind(ModuleKind.ESModule)
)
`
But I need to have a forensic look at what I missed / is different, because I'm annoyed with myself. I'm sorry for the time I cost you.

**Comment by Quafadas** [comment]
I'm terribly sorry and thanks for your patience. It was exactly the line above. I believe it altered the behaviour of mdocs linker. 

In hindsight, that's actually pretty reasonable behaviour, it also explains why I couldn't get anywhere - I was essentially trying to solve the wrong problem :-(. 

This now works really well. Thankyou so much again.

**Pull request closed without merging**